eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,173 @@
+
+                DetailAST identToken = ast.findFirstToken(TokenTypes.IDENT);
+                if (identToken == null) {
+                    return;
+                }
+
+                String newClientLogger = identToken.getText();
+                if (!newClientLogger.equals(CLIENT_LOGGER)) {
+                    return;
+                }
+
+                DetailAST elistToken = ast.findFirstToken(TokenTypes.ELIST);
+                DetailAST exprToken = elistToken.findFirstToken(TokenTypes.EXPR);
+                if (exprToken.getFirstChild().getType() != TokenTypes.DOT) {
+                    return;
+                }
+                String containerClassName = FullIdent.createFullIdentBelow(exprToken).getText();
+                containerClassName = containerClassName.substring(0, containerClassName.length() - 6);
+                if (!containerClassName.equals(className)) {
+                    log(ast, String.format("Not newing a ClientLogger with matching class name. Use ''%s.class'' instead of ''%s.class''", className, containerClassName));
+                }
+                break;
+            case TokenTypes.VARIABLE_DEF:
+                if (!hasClientLoggerImported || !isTypeClientLogger(ast)) {
+                    return;
+                }
+                // Check if the Logger instance named as 'logger'.
+                DetailAST identAST = ast.findFirstToken(TokenTypes.IDENT);
+                if (identAST != null && !identAST.getText().equals(loggerRequiredName)) {
+                    log(ast, String.format(LOGGER_NAME_ERROR, loggerRequiredName, identAST.getText()));
+                }
+                // Check if the Logger is static instance, log as error if it is static instance logger.
+                DetailAST modifierAST = ast.findFirstToken(TokenTypes.MODIFIERS);
+                if (modifierAST != null && modifierAST.branchContains(TokenTypes.LITERAL_STATIC)) {
+                    log(ast, STATIC_LOGGER_ERROR);
+                }
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * check if the VARIABLE_DEF AST node type is 'ClientLogger'.
+     * @param varDefAST VARIABLE_DEF AST node
+     * @return true if the variable type is 'ClientLogger'.
+     */
+    private boolean isTypeClientLogger(DetailAST varDefAST) {
+        DetailAST typeAST = varDefAST.findFirstToken(TokenTypes.TYPE);
+        DetailAST varNameAST = varDefAST.findFirstToken(TokenTypes.IDENT);
+        if (typeAST == null || varNameAST == null) {
+            return false;
+        }
+        DetailAST typeNameAST = typeAST.findFirstToken(TokenTypes.IDENT);
+        if (typeNameAST == null) {
+            return false;
+        }
+        String typeName = typeNameAST.getText();
+        if (typeName != null && typeName.equals(CLIENT_LOGGER)) {
+            return true;
+        }
+        return false;
```

Comment:
This can be condensed into `return typeName != null && typeName.equals(CLIENT_LOGGER);`